Objects/typeobject.c

```diff
@@ -901,6 +901,21 @@ type_repr(PyTypeObject *type)
 }
 
 static PyObject *
+type_str(PyTypeObject *type)
+{
+    PyObject *name, *rtn;
+
+    name = type_qualname(type, NULL);
```

Comment:
Wouldn't be better to include a module name for getting rid of ambiguity?

I would just return `PyUnicode_FromString(type->tp_name)`.

Reply:
The original discussion on the ticket mentioned only the type qualname, without its module.  Looking at Lib/typing.py and other files (for example the error message in functools), I think maybe including the module name too would be useful.  Let’s see comments after I get a smaller patch that passes the test suite.